server/catgenome/src/main/java/com/epam/catgenome/manager/heatmap/HeatmapManager.java

```diff
@@ -0,0 +1,362 @@
+        return file;
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void updateLabelAnnotation(final long heatmapId, final String path) throws IOException {
+        Assert.isTrue(!TextUtils.isBlank(path), MessagesConstants.PATH_IS_REQUIRED);
+        File file = getFile(path);
+        Heatmap heatmap = heatmapDao.loadHeatmap(heatmapId);
+        Assert.notNull(heatmap, getMessage(MessagesConstants.ERROR_HEATMAP_NOT_FOUND, heatmapId));
+        checkLabelAnnotation(path, heatmap.getRowLabels(), heatmap.getColumnLabels());
+        heatmapDao.updateLabelAnnotation(heatmapId, FileUtils.readFileToByteArray(file), path);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void updateCellAnnotation(final long heatmapId, final String path) throws IOException {
+        Assert.isTrue(!TextUtils.isBlank(path), MessagesConstants.PATH_IS_REQUIRED);
+        File file = getFile(path);
+        Heatmap heatmap = heatmapDao.loadHeatmap(heatmapId);
+        Assert.notNull(heatmap, getMessage(MessagesConstants.ERROR_HEATMAP_NOT_FOUND, heatmapId));
+        checkCellAnnotation(path, heatmap.getRowLabels(), heatmap.getColumnLabels());
+        heatmapDao.updateCellAnnotation(heatmapId, FileUtils.readFileToByteArray(file), path);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void deleteHeatmap(final long heatmapId) {
+        Heatmap heatmap = heatmapDao.loadHeatmap(heatmapId);
+        Assert.notNull(heatmap, getMessage(MessagesConstants.ERROR_HEATMAP_NOT_FOUND, heatmapId));
+        heatmapDao.deleteHeatmap(heatmapId);
+        biologicalDataItemManager.deleteBiologicalDataItem(heatmap.getBioDataItemId());
+    }
+
+    public Heatmap getHeatmap(final long heatmapId) {
+        final Heatmap heatmap = heatmapDao.loadHeatmap(heatmapId);
+        Assert.notNull(heatmap, getMessage(MessagesConstants.ERROR_HEATMAP_NOT_FOUND, heatmapId));
+        return heatmap;
+    }
+
+    public List<List<Map<?, String>>> getContent(long heatmapId) throws IOException {
+        Heatmap heatmap = heatmapDao.loadHeatmap(heatmapId);
+        Assert.notNull(heatmap, getMessage(MessagesConstants.ERROR_HEATMAP_NOT_FOUND, heatmapId));
+        final String separator = getSeparator(heatmap.getPath());
+        InputStream heatmapInputStream = heatmapDao.loadHeatmapContent(heatmapId);
+        InputStream annotationInputStream = heatmapDao.loadCellAnnotation(heatmapId);
+        return getAnnotatedContent(heatmapInputStream, annotationInputStream, heatmap.getCellValueType(), separator);
+    }
+
+    public List<List<String>> getLabelAnnotation(final long heatmapId) throws IOException {
+        Heatmap heatmap = heatmapDao.loadHeatmap(heatmapId);
+        Assert.notNull(heatmap, getMessage(MessagesConstants.ERROR_HEATMAP_NOT_FOUND, heatmapId));
+        final String separator = getSeparator(heatmap.getLabelAnnotationPath());
```

Comment:
If annotation file is not specified this method should return empty result (at the moment it throws and error)